File: demo_app/lib/SecretKeeper.dart
import 'package:flutter/material.dart';

class SecretKeeper extends StatefulWidget {
  @override
  State<StatefulWidget> createState() {
    return _SecretKeeperState();
  }
}

class _SecretKeeperState extends State<SecretKeeper> {
  List<String> secrets = [
    "I picked my nose multiple times during Zoom calls",
    "I've bought some toilet paper in advance",
    "I've secretly been fedding the mouse in our cellar",
    "I vibe to Justin Bieber's 'Baby'",
    "Sometimes I wish I was born as a squirrel",
    "To kill time during isolation, I've bought a Wii U.",
    "My Airpod fakes from AliExpress were confiscated by customs",
    "My favorite T-Shirt has a Dugong on it"
  ];

  final TextEditingController eCtrl = new TextEditingController();

  Widget buildSecretCard(String text) {
    return Padding(
      padding: EdgeInsets.symmetric(vertical: 10),
      child: Card(
        child: Padding(
          padding: EdgeInsets.all(20),
          child: Text(
            text,
            style: TextStyle(fontSize: 18),
          )
        )
      ),
    );
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
        appBar: AppBar(title: Text("Your secret secrets")),
        body: Padding(
            padding: EdgeInsets.symmetric(horizontal: 16),
            child: Column(
              children: <Widget>[
                TextField(
                    controller: eCtrl,
                    onSubmitted: (text) {
                      setState(() {
                        secrets.insert(0, text);
                      });
                      eCtrl.clear();
                    },
                    decoration: InputDecoration(
                      border: UnderlineInputBorder(),
                      labelText: "Add a secret ...",
                      labelStyle: TextStyle(fontSize: 18)
                    ),
                  ),
                Padding(padding: EdgeInsets.all(10)),
                Expanded(
                  child: ListView.builder(
                      itemCount: secrets.length,
                      itemBuilder: (context, index) {
                        return buildSecretCard(secrets[index]);
                      }),
                ),
              ],
            )));
  }
}

File: new_demo_app/Reverse Engineering Mobile Apps Talk Demo App/Reverse Engineering Mobile Apps Talk Demo App/SecretKeeper.swift
//
//  SecretKeeper.swift
//  Reverse Engineering Mobile Apps Talk Demo App
//
//  Created by Simon Knott on 18.04.20.
//  Copyright © 2020 Simon Knott. All rights reserved.
//

import SwiftUI

struct SecretKeeper: View {
    
    @State var input: String = ""
    
    @State var secrets: [String] = [
        "I picked my nose multiple times during Zoom calls",
        "I've bought some toilet paper in advance",
        "I've secretly been fedding the mouse in our cellar",
        "I vibe to Justin Bieber's 'Baby'",
        "Sometimes I wish I was born as a squirrel",
        "To kill time during isolation, I've bought a Wii U.",
        "My Airpod fakes from AliExpress were confiscated by customs",
        "My favorite T-Shirt has a Dugong on it"
    ]
    
    func sneakilyUploadSecretToServer() {
        
    }
    
    func handleCommit() {
        sneakilyUploadSecretToServer()
        
        secrets.insert(input, at: 0)
        input = ""
    }
    
    var body: some View {
        VStack {
            TextField(
                "Add a secret ...",
                text: $input,
                onCommit: handleCommit
            )
                .multilineTextAlignment(.center)
            
            List(secrets, id: \.hashValue) { secret in
                Text(secret)
            }
        }
        .navigationBarTitle("Secret Keeper")
    }
}

struct SecretKeeper_Previews: PreviewProvider {
    static var previews: some View {
        NavigationView {
            SecretKeeper()
        }
    }
}
